### PR TITLE
Remove treatment implementation

### PR DESCRIPTION
**Pull request by dippindots**
The fifth step of https://github.com/cBioPortal/cbioportal/issues/6769.

- Removed treatment controller, services, tests, importer and related files.
- Keep the data file, transfer the data file to the generic_assay file.
- Remove treatment validator

### CODE REVIEW

File: core/src/test/scripts/unit_tests_validate_data.py [hunk 1]
```diff
@@ -1027,168 +1027,6 @@ def test_comma_in_feature_id_issues_error(self):
...
-    # create a dummy class of that has no value checking
-    class DummyTreatmentValidator(validateData.TreatmentWiseFileValidator):
-        def checkValue(self, value, col_index):
-            return
-
-    def setUp(self):
-        super().setUp()
-        # reset state of the TreatmentWiseFileValidator class 
-        # to prevent carry-over from other tests
-        _resetMultipleFileHandlerClassVars()
```
Comment (dippindots) [inline core/src/test/scripts/unit_tests_validate_data.py]: Should bring back those tests for `treatment data` importing if needed.

File: core/src/main/scripts/importer/validateData.py [hunk 2]
```diff
@@ -3895,81 +3892,6 @@ def get_message_features_do_not_match(cls):
...
-    def get_prior_validated_sample_ids():
-        return TreatmentWiseFileValidator.prior_validated_sample_ids
-
-    @staticmethod
-    def set_prior_validated_sample_ids(sample_ids):
-        TreatmentWiseFileValidator.prior_validated_sample_ids = sample_ids
-
-    @classmethod
-    def get_message_features_do_not_match(cls):
-        return "Treatment feature columns (`entity_stable_id`, ...) in treatment profile data files are not identical. The same set of treatments should be used across the different treatment data files for this study. Please ensure that all entity stable id's of one file are present in all other treatment files."
```
Comment (dippindots) [inline core/src/main/scripts/importer/validateData.py]: Should bring back those tests for `treatment data` importing if needed.

File: core/src/test/scripts/unit_tests_validate_data.py [hunk 3]
```diff
@@ -1027,168 +1027,6 @@ def test_comma_in_feature_id_issues_error(self):
...
-        passConcValue("10.23234").warning.assert_not_called()
-        passConcValue("10.23234").error.assert_not_called()
-
-# helper function for test 
-def passConcValue(value):
-    mockval = Mock()
-    validateData.TreatmentValidator.checkValue(mockval, value, 1)
-    return mockval.logger
-
-# ------------------------ end treatment validator test -------------------------
```
Comment (dippindots) [inline core/src/test/scripts/unit_tests_validate_data.py]: Should bring back those tests for `treatment data` importing if needed.

File: core/src/main/scripts/importer/validateData.py [hunk 4]
```diff
@@ -3895,81 +3892,6 @@ def get_message_features_do_not_match(cls):
...
-
-
```
Comment (dippindots) [inline core/src/main/scripts/importer/validateData.py]: Should bring back those tests for `treatment data` importing if needed.

**Comment by dippindots** [comment]
Hi @pvannierop, all the `treatment` pr is ready now, could you review them when you got a chance? Thanks! One thing please note: @n1zea144 and I had discussions the validation of the `generic_assay` file, we think for `generic_assay` file, we will not require all files (although some may have the same `generic_assay_type`) contain the same properties or the same entities. I have logged an issue here for discussion: https://github.com/cBioPortal/cbioportal/issues/6826. Thanks again!

File: core/src/main/scripts/importer/validateData.py [hunk 5]
```diff
@@ -5072,8 +4915,7 @@ def main_validate(args):
                                          hugo_entrez_map=None,
                                          alias_entrez_map=None,
                                          gene_set_list=None,
-                                         gene_panel_list=None,
-                                         treatment_map=None)
```
Comment (pvannierop) [inline core/src/main/scripts/importer/validateData.py]: Are you aware that you have eliminated the existing mechanism that checks for and acts upon entities with the same name being present in the database! How is this currently handled for Generic Assay entities? When this is not handled in the validator, the user is not informed of the consequences of import of the current dataset.

File: core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java [hunk 6]
```diff
@@ -117,21 +117,14 @@ public static GeneticProfile loadGeneticProfile(File file) throws IOException, D
...
-            geneticProfile.setPivotThreshold( Float.parseFloat( geneticProfile.getOtherMetaDataField("pivot_threshold_value") ) );
-            geneticProfile.setSortOrder( geneticProfile.getOtherMetaDataField("value_sort_order") );
-        }
 
         if (geneticProfile.getGeneticAlterationType() == GeneticAlterationType.GENERIC_ASSAY) {
             validateGenericAssay(geneticProfile, file);
             geneticProfile.setGenericAssayType(geneticProfile.getOtherMetaDataField("generic_assay_type"));
-            // TODO: if genericAssayType is TREATMENT_RESPONSE, validate and set pivotThreshold and sortOrder
-            if (geneticProfile.getOtherMetaDataField("pivot_threshold_value") != null) {
+            // if genericAssayType is TREATMENT_RESPONSE, validate and set pivotThreshold and sortOrder
```
[the remark this answers is not in the document]
Comment (dippindots) [reply core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java]: @pvannierop Do you know why we cannot validate those fields in the python validator? I am not quite sure about it.

File: core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java [hunk 7]
```diff
@@ -117,21 +117,14 @@ public static GeneticProfile loadGeneticProfile(File file) throws IOException, D
...
+            if (geneticProfile.getGenericAssayType().equals("TREATMENT_RESPONSE")) {
```
[the remark this answers is not in the document]
Comment (dippindots) [reply core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java]: Sure, we can keep it for every profile which has those fields. But I am not sure if that should be required. If I just want to use the `generic_assay` data in the `patient view`, I don't think those fields are required. Any thoughts @n1zea144?

File: core/src/main/scripts/importer/validateData.py [hunk 8]
```diff
@@ -3824,22 +3822,21 @@ def parseFeatureColumns(self, nonsample_col_vals):
...
+            # Check if this is the second data file
             if self.get_prior_validated_feature_ids() is not None:
-                # Check if treatment is in the first treatment data file
+                # Check if feature is in the first data file
                 if feature_id not in self.get_prior_validated_feature_ids():
                     self.logger.error('Feature id cannot be found in other data file',
                                       extra={'line_number': self.line_number,
                                              'cause': feature_id})
-            # Add treatment to list of treatments of current treatment data file
+            # Add feature to list of features of current data file
```
[the remark this answers is not in the document]
Comment (dippindots) [reply core/src/main/scripts/importer/validateData.py]: Sure, will do!

File: core/src/main/scripts/importer/validateData.py [hunk 9]
```diff
@@ -3895,81 +3892,6 @@ def get_message_features_do_not_match(cls):
         return "Gene sets column in score and p-value file are not equal. The same set of gene sets should be used in the score and p-value files for this study. Please ensure that all gene set id's of one file are present in the other gene set data file."
 
 
-class TreatmentWiseFileValidator(MultipleDataFileValidator, metaclass=ABCMeta):
```
[the remark this answers is not in the document]
Comment (dippindots) [reply core/src/main/scripts/importer/validateData.py]: Yes, I logged this in this issue for discussion: https://github.com/cBioPortal/cbioportal/issues/6826. I have already made an approach to bring those tests back, I will send you the link soon. Thanks!

File: core/src/main/scripts/importer/validateData.py (hunk 5 above)
Comment (dippindots) [reply core/src/main/scripts/importer/validateData.py]: Same to the previous comment.

File: core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java (hunk 7 above)
[the remark this answers is not in the document]
Comment (pvannierop) [reply core/src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java]: @dippindots These are not required fields. At present default values of `ASC` and `0` are  put in the database for Treatment response profiles. We can extend this strategy to Generic assay. The reason for me bringing this up is that I foresee that users typically would like to represent Generic Assay data with the Waterfall plot type. For this the `pivot_threshold` and `sort_order` are required. So, my proposed strategy would be make the `pivot_threshold` and `sort_order`  fields optional, but always create a fall back value when not specified. This way the user can always decide to represent the profile in Plots Tab.

**Comment by pvannierop** [comment]
@n1zea144 @dippindots We may need a more indepth discussion on how treatment response files were validated and how this is related to the new generic endpoint implementation. From File-Formats.md: 

> When multiple treatment data files are loaded as part of study, the `entity_stable_id`, `META:name`, `META:description` and `META:url` columns must be identical is all data files.

This requirement is enforced in the validator. We need to investigate/discuss how this works for the generic assay data.

**Comment by n1zea144** [comment]
@pvannierop @inodb @dippindots We will merge this PR when frontend master to rc is complete. 
 Will continue discussion of treatment validation on PR [6841](https://github.com/cBioPortal/cbioportal/pull/6841)

**Pull request merged**